### PR TITLE
open DescriptionElements to handle ActivityElements

### PR DESCRIPTION
**Pull request by camhashemi**
@leogrim @ryanpbrewster

### CODE REVIEW

**Comment by camhashemi** [comment]
this is now a subset of #20530, but perhaps worth merging separately first

File: kifi-backend/modules/common/app/com/keepit/common/util/ActivityElements.scala [hunk 1]
```diff
@@ -41,15 +44,15 @@ object ActivitySource extends Enumerator[ActivitySource] {
   implicit val writes: Writes[ActivitySource] = Writes { o => JsString(o.value) }
 }
 
-sealed abstract class ActivityElement extends DescriptionElement
+trait ActivityElement extends DescriptionElement
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/common/util/ActivityElements.scala:47]: Do you need the `ActivityElement` trait at all now? Isn't it just `DescriptionElement`?

Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/common/util/ActivityElements.scala:47]: And move all of these to `DescriptionElements.scala` and keep the trait sealed?

File: kifi-backend/modules/common/app/com/keepit/common/util/ActivityElements.scala [hunk 2]
```diff
@@ -106,15 +128,21 @@ object ActivityElement {
...
+  implicit def fromUser(bu: BasicUser): UserElement = UserElement(bu.externalId, bu.fullName, bu.pictureName, Path(bu.username.value))
+  implicit def fromBasicOrg(bo: BasicOrganization): OrganizationElement = OrganizationElement(bo.orgId, bo.name, bo.avatarPath.path, bo.path)
+
+  implicit def fromNonUser(bnu: BasicNonUser): NonUserElement = NonUserElement(bnu.id)
+  implicit def fromBasicAuthor(ba: BasicAuthor): AuthorElement = ba match {
+    case KifiUser(id, name, picture, url) => AuthorElement(id, name, picture, Some(url))
+    case SlackUser(id, name, picture, url) => AuthorElement(id, name, picture, Some(url))
+    case TwitterUser(id, name, picture, url) => AuthorElement(id, name, picture, Some(url))
+  }
+  implicit def fromBasicLibrary(bl: BasicLibrary): LibraryElement = LibraryElement(bl.id, bl.name, bl.color, Path(bl.path))
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/common/util/ActivityElements.scala:140]: Sweet!

File: kifi-backend/modules/common/app/com/keepit/common/util/ActivityElements.scala [hunk 3]
```diff
@@ -106,15 +128,21 @@ object ActivityElement {
...
 
   implicit val writes = Writes[ActivityElement] {
-    case txt: TextElement => TextElement.writes.writes(txt)
     case usr: UserElement => UserElement.writes.writes(usr)
     case nur: NonUserElement => NonUserElement.writes.writes(nur)
+    case aut: AuthorElement => AuthorElement.writes.writes(aut)
     case lib: LibraryElement => LibraryElement.writes.writes(lib)
     case org: OrganizationElement => OrganizationElement.writes.writes(org)
   }
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/common/util/ActivityElements.scala:148]: Check out `SlackAuthenticatedAction` for a slightly cleaner way to write these (use `def`).

**Comment by leogrim** [comment]
A couple of small comments, neat!

**Pull request merged**